docs/book/src/plugins/go-v3-plugin.md

```diff
@@ -0,0 +1,22 @@
+# go/v3
+
+This is the default plugin that is used by Kubebuilder to construct sets of [controllers](https://github.com/kubernetes-sigs/controller-runtime). It basically scaffolds all the boilerplate code required to create and design controllers. Check the following [example](https://github.com/kubernetes-sigs/kubebuilder/tree/master/testdata/project-v3) to view the files created or updated by this plugin when used with the supported subcommands. You can also follow the [Building CronJob](cronjob-tutorial.md) tutorial to get a step-by-step explanation of how the plugin works with different subcommands.
+
+## How to use it ?
+As go/v3 is the default plugin there is no need to explicitly mention to Kubebuilder to use this plugin. 
+To create a new project with the go/v3 plugin the following command can be used: 
+```sh
+kubebuilder init --domain tutorial.kubebuilder.io --repo tutorial.kubebuilder.io/project
+```
+All the other subcommands supported by the go/v3 plugin can be executed similarly.
+
+If needed we can explicitly tell Kubebuilder to use the go/v3 plugin by adding ```--plugins=go/v3```.
+
+## Subcommands supported by the plugin
+-  Init -  `kubebuilder init [OPTIONS]`
+-  Edit -  `kubebuilder edit [OPTIONS]`
+-  Create API -  `kubebuilder create api [OPTIONS]`
+-  Create Webhook - `kubebuilder create webhook [OPTIONS]`
+
+## Further resources
+- For full implementation of this plugin check out [go/v3](https://pkg.go.dev/sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v3).
```

Comment:
I liked that we added the go docs. But it is not the code implementation.
So, I'd recommend something like :  

- Check in the [main.go](https://github.com/kubernetes-sigs/kubebuilder/blob/master/cmd/main.go) how the plugin is composited
- Check the code implementation for the base.go.kubebuilder.io/v3
- Check the code implementation for  kustomize.common.kubebuilder.io/v1
- Check the [controller-runtime] to know more about controllers